core/arch/aarch64/instr_create.h

```diff
@@ -445,6 +436,16 @@
 #define INSTR_CREATE_bl(dc, pc) \
   instr_create_1dst_1src((dc), OP_bl, opnd_create_reg(DR_REG_X30), (pc))
 
+#define INSTR_CREATE_adc(dc, Rd, Rn, Rm) \
+  instr_create_1dst_2src((dc), OP_adc, (Rd), (Rn), (Rm))
+#define INSTR_CREATE_adcs(dc, Rd, Rn, Rm) \
+  instr_create_1dst_2src((dc), OP_adcs, (Rd), (Rn), (Rm))
+#define INSTR_CREATE_adds_extend(dc, Rd, Rn, Rm, shift, imm3) \
+  instr_create_1dst_4src((dc), OP_adds, (Rd), (Rn), opnd_create_reg_ex(opnd_get_reg(Rm), 0, DR_OPND_EXTENDED), opnd_add_flags((shift), DR_OPND_IS_EXTEND), (imm3))
+#define INSTR_CREATE_adds_imm(dc, Rd, Rn, imm12, shift_amt) \
```

Comment:
Is the change in behaviour for `INSTR_CREATE_adds_shift` intentional? Maybe we should at least document the change in behaviour?

Reply:
Yes, intent was to make the adds variants more explicit.

Reply:
Sounds good to me.